### PR TITLE
ZM0sHy6Z setup server

### PR DESCRIPTION
**Pull request by Niyiragira**
**What does this PR do?** 
Requests to merge ch-setup-server-welcome-ZM0sHy6Z with develop

 **Description of Task to be completed?**
Adding basic chores and a running server with ES6 code
Has these completed:
1.Setup the project structure
2.A welcome end point is set up
3.Configured Eslint
4.Setup API documentation with Swagger

**How should this be manually tested?**

- clone the repo
-do git checkout origin/ch-setup-server-welcome-ZM0sHy6Z
-do git init and git install
-run npm run dev

- use postman http://localhost:5000/phantom-docs/  for a GET request

- Browse http://localhost:5000/phantom-docs/    for swagger documentation of a GET request

**What are the relevant pivotal tracker stories?**

https://trello.com/c/FMtr8tcx/23-set-up-nodejs-rest-api-empty-project
**Screenshots**
Screen Shot 2020-09-24 at 09 40 51

### CODE REVIEW

**Review by ncutixavier — DISMISSED** [review]
You did good for the code. But next time you have to write a good PR that is easy to read like inserting a space at the end of paragraph.

**Comment by Nkbtemmy** [comment]
you did great job but your PR description is not describing well how it will be reviewed; try to improve and also i ask you if you may put project setup in folder

**Review by Nkbtemmy — DISMISSED** [review]

**Review by karigirwa — APPROVED** [review]

**Review by ncutixavier — APPROVED** [review]
- Try to change the extension of eslint to json, it will make easy to read.
- Change swagger to another format. 
reference: https://github.com/ncutixavier/home/blob/develop/swagger.json

**Comment by Nkbtemmy** [comment]
Again you did great job but there same missing information in your PR description of how people will test your work (like you don't mention how reviews will test Swagger documentation ) and you are saying that we have to clone your branch that will not be correct because you clone repo not branch and you cannot clone repo witch is not merged

**Comment by Niyiragira** [comment]
> Again you did great job but there same missing information in your PR description of how people will test your work (like you don't mention how reviews will test Swagger documentation ) and you are saying that we have to clone your branch that will not be correct because you clone repo not branch and you cannot clone repo witch is not merged

Thanks the issue have been corrected

**Review by Nkbtemmy — APPROVED** [review]

File: .babelrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "presets": ["@babel/preset-env"]
+}
```
Comment (blaise82) [inline .babelrc:3]: add a new line

File: src/app.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import routes from "./routes/index";
+import welcomeRoute from "./routes/welcome"
+
+app.use(express.json());
+app.use(routes)
+
+app.use("/welcomes",welcomeRoute);
+
+
+// eslint-disable-next-line no-undef
```
Comment (blaise82) [inline src/app.js:13]: do not disable any eslint rules

File: src/app.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+
+import express from "express";
+const app=express();
+import routes from "./routes/index";
+import welcomeRoute from "./routes/welcome"
+
```
Comment (blaise82) [inline src/app.js:6]: add body parser express

File: src/app.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+app.use(express.json());
+app.use(routes)
+
+app.use("/welcomes",welcomeRoute);
+
+
+// eslint-disable-next-line no-undef
+const PORT =process.env.PORT || 5000;
+
+app.listen(PORT, console.log(`Server started on port ${PORT}`))
```
Comment (blaise82) [inline src/app.js:16]: please make sure that your eslint is working properly
to avoid the new line issue

File: src/app.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+
+import express from "express";
+const app=express();
+import routes from "./routes/index";
+import welcomeRoute from "./routes/welcome"
+
+app.use(express.json());
+app.use(routes)
+
+app.use("/welcomes",welcomeRoute);
```
Comment (blaise82) [inline src/app.js:10]: you shouldn't be having this in the app.js file
it should be in the routes folder

File: src/utils/swagger.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+const swaggerOptions = {
+    swaggerDefinition: {
+      info: {
+        version: "1.0.0",
+        title: "Phantom API",
+        description: "Phantom API Information",
+        contact: {
+          name: "Phantom project documentation"
+        },
+        servers: ["http://localhost:5000"]
```
Comment (blaise82) [inline src/utils/swagger.js:12]: add the Heroku server

File: src/utils/swagger.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+      }
+    },
+  
+    // eslint-disable-next-line no-undef
+    apis: [path.resolve(__dirname,'../routes/*.js')]
+  };
+
+  export default swaggerOptions;
```
Comment (blaise82) [inline src/utils/swagger.js:20]: add a  new line

**Review by blaise82 — CHANGES_REQUESTED** [review]
Please do change your PR title to the proper format and improve your PR description

**Comment by blaise82** [comment]
make sure that you have one commit

**Pull request closed without merging**